internal/pkg/istio-router-check/cmd/root.go

```diff
@@ -0,0 +1,186 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/getyourguide/istio-config-validator/internal/pkg/istio-router-check/envoy"
+	"github.com/getyourguide/istio-config-validator/internal/pkg/istio-router-check/helpers"
+	"github.com/go-logr/logr"
+	"github.com/spf13/cobra"
+	"istio.io/istio/pkg/util/protomarshal"
+)
+
+const envoyRouterCheckTool string = "router_check_tool"
+
+type RootCommand struct {
+	ConfigDir               string
+	TestDir                 string
+	Details                 bool
+	DisableDeprecationCheck bool
+	OnlyShowFailures        bool
+	FailThreshold           float64
+	CoverageAll             bool
+	OutputDir               string
+	DetailedCoverage        bool
+	Verbosity               int
+}
```

Comment:
Would it be worth nesting the flags that are specifically for the `routerCheckFlags` (so we group them under one field). It could make it clearer which flags are for this binary, and which are for the called binary.

Reply:
Makes sense.